### ISSUE TITLE
`Ctrl+C` doesn't stop `yarn upgrade`

### ISSUE THREAD

**Issue by wescoder** [yarnpkg/yarn#4982]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
`Ctrl+C` does not terminate `yarn upgrade`
**If the current behavior is a bug, please provide the steps to reproduce.**
Run `yarn upgrade` and press `Ctrl+C`

**What is the expected behavior?**
`Ctrl+C` should stop the command.

**Please mention your node.js, yarn and operating system version.**
Yarn: v1.3.2
Node: v8.8.1
OS: Linux 4.9.0-deepin13-amd64 #1 SMP PREEMPT Deepin 4.9.57-1 (2017-10-19) x86_64 GNU/Linux

**Comment by rally25rs** [comment]
This should behave exactly the same as the rest of Yarn. It seems to work fine for me on OSX Sierra using Bash shell:

```
~/Projects/yarn-test : yarn upgrade
yarn upgrade v1.3.2
[1/4] 🔍  Resolving packages...
⠂ inherits@~2.0.1^C

~/Projects/yarn-test : yarn upgrade
yarn upgrade v1.3.2
[1/4] 🔍  Resolving packages...
⠁ has-flag@^2.0.0^C

~/Projects/yarn-test : yarn upgrade
yarn upgrade v1.3.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[---------------------------------------------------------------------------------------------------------------------------------------------------------------------] 0/424^C
```

I don't see any Yarn code explicitly handing `SIGINT` (we do capture `SIGTERM` though) so I suspect this should just kill Node.js at it's handler for that signal.

I don't have a Linux VM handy to test on right now. I can give it a try on Debian this evening.

If you want to dig-in and help debug, you could try adding a `SIGINT` handler to `src/util/signal-handler.js` and see if it is ever called when `ctrl-c` is pressed.

**Comment by wescoder** [comment]
Thank you for looking into it.
I've found earlier today (3 hours ago) that it had to be something into my shell.

I had to re-install my system (due to an unrelated conflict of my graphic card and kernel) and this bug that annoyed me for months is finally away.
I think this potentially is not affecting other people and is very specific.
This could be caused by anything in my 200 LOC `.zshrc` or my smaller 50 LOC `.bashrc`, or maybe any other shell enhancers on my old system.

I'll be alert into figuring out what was causing this issue while installing my tools into my new system and if i find something useful I'll comment here in order to help anyone that may possible pass through the same.

For now I'm closing the issue.
Thanks a lot for your time.
